libqtile/widget/groupbox.py

```diff
@@ -127,7 +128,7 @@ def drawbox(
         if block and bordercolor is not None:
             framed.draw_fill(offset, y, rounded)
         elif line:
-            framed.draw_line(offset, y, highlighted)
+            framed.draw_line(offset, y, highlighted, toggle=toggle)
```

Comment:
Couldn't you just check the state of `highlighted` and `toggle` here and set the `y` value accordingly, rather than adding code to `drawer.py`?

The `toggle_line` name is a bit confusing too. You seem to be changing the position of the line rather than the visibility of the line.

Reply:
I think I could do it, but I'm not sure if it will apply when highlighted, since `highlight` is inside drawer.py

Reply:
And yes, the name is a bit confusing, how would you name it?

Reply:
> I think I could do it, but I'm not sure if it will apply when highlighted, since `highlight` is inside drawer.py

Isn't the `highlight` in `drawer.py` just the `highlighted` argument in the `draw_line` call above? If so, you can do all the condition checking here.

Reply:
Yes, that's right, I'll try it.

Reply:
Ok, I can't set a value to `y` from `groupbox.py` because in `drawer.py` the value of `y` is always `self.height - self.border_width`.

Reply:
@elParaguayo so?

Reply:
So you can use `framed.height` and `framed.border_width`.

I will confess, I'm finding this whole PR a bit confusing. The `invert` parameter is just to put the line on the top for groups on the inactive screen. Is that right? 

Conceivably, somebody might want that the other way round. So, maybe the text frame needs to be more flexible as to where the line is drawn (and this widget should also be updated to work on vertical bars so we would need lines on left and right).

If that's sounding like too much work then you may want to keep your changes within your config and wait until I get round to doing the other bits.

Reply:
> So you can use `framed.height` and `framed.border_width`.

But if I modify those values, won't it affect the border width and bar size?

> The invert parameter is just to put the line on the top for groups on the inactive screen. Is that right?

Yes.

> Conceivably, somebody might want that the other way round.

That's easy to fix, I just need to create another argument or create just one that contains 3 possible values: `'None'`, `'Up'` and `'Down'`.

> this widget should also be updated to work on vertical bars

Huh?, the docs says that it only supports horizontal orientation.

Reply:
Never mind, closed for now while the rest is completed.

Reply:
> Huh?, the docs says that it only supports horizontal orientation.

Yes, it does. My point was that, there's no reason this widget couldn't work on a vertical bar too and, once that change has been made, that would have implications for this PR too.

Reply:
Yup, I understand it :)